game-core/src/main/java/games/strategy/engine/framework/startup/ui/InGameLobbyWatcher.java

```diff
@@ -113,7 +117,12 @@ private InGameLobbyWatcher(
             .gameVersion("0")
             .build();
 
-    gameId = gameToLobbyConnection.postGame(gameDescription.toLobbyGame());
+    final PostGameResponse postGameResponse =
+        gameToLobbyConnection.postGame(gameDescription.toLobbyGame());
+
+    gameId = postGameResponse.getGameId();
+
+    connectivityHandler.accept(postGameResponse.isSuccessfulConnectivityCheck());
```

Comment:
The connectivity handler perhaps should only be invoked on a failed connectivity check. I think it would create a much tighter abstraction rather than leaking details of everything that is involved with connecting. EG:

```
if(postGamREsponse.getErrorCode() == CONNECTIVITY_CHECK_FAILED) {
  connectivityHandler.accept( ... error message ... );
}
```

I'm also a bit concerned that the gameId is in an odd state in this case as well. FWIW this is where the code flow is probably simpler if we have the two be separate.

EG:

```
if(!connectvitiyCheck.verify( ..)) {
    connectivityHandler.accept( ... error message ... );
}

gameId = gameToLobbyConnection.postGame(gameDescription.toLobbyGame());

```